modules/apps/static/portal-dependency-manager-component-executor-factory/portal-dependency-manager-component-executor-factory/src/main/java/com/liferay/portal/dependency/manager/component/executor/factory/internal/DependencyManagerSyncImpl.java

```diff
@@ -135,10 +144,48 @@ public void complete(Future<Void> future) {
 			});
 	}
 
+	private void _registerDependencyManagerSyncGatekeeperCallable() {
+		registerSyncCallable(
+			() -> {
+				String componentName =
+					DependencyManagerSyncGatekeeper.class.getName();
+
+				CountDownLatch countDownLatch = new CountDownLatch(1);
+
+				BundleContext bundleContext =
+					_componentContext.getBundleContext();
+
+				ServiceListener serviceListener = serviceEvent -> {
+					if (serviceEvent.getType() != ServiceEvent.REGISTERED) {
+						return;
+					}
+
+					bundleContext.getService(
```

Comment:
@yuhai @dantewang Why this getService is needed?

Reply:
@tinatian, getService() means DependencyManagerSyncGatekeeper service activate() finished. And then runs the following logic.